src/com/zone5cloud/retrofit/core/OkHttpClientInterceptor_Authorization.java

```diff
@@ -195,33 +210,38 @@ private void refreshIfRequired(Chain chain, String baseUrl) {
 				token = this.token.get();
 				if (token != null && token.isExpired()) {
 					try {
-						if (token.getRefreshToken() != null) {
-							// do a Cognito refresh
-							String username = token.extractUsername();
-							if (username != null) {
-								RequestBody body = new FormBody.Builder().add("client_id", clientID)
-																		 .add("client_secret", clientSecret)
-																		 .add("grant_type", GrantType.REFRESH_TOKEN.toString())
-																		 .add("username", username)
-																		 .add("refresh_token", token.getRefreshToken()).build();
-								String url = baseUrl + Users.NEW_ACCESS_TOKEN;
-								Request authRequest = new Request.Builder().url(url).post(body).build();
+							if (token.getRefreshToken() != null) {
+								// do a Cognito refresh
+								String username = token.extractUsername();
+//
+								if (username != null) {
```

Comment:
Instead of extracting the username from the token, use `this.username`, which is passed into the constructor with ClientConfig. This is the reason that we added username to the config. We can no longer rely on it being in the token.

Reply:
There were more than one usages of it hence put it into a variable but I can revert it and use this.userName twice